src/renderer/components/outputs/DirectoryOutput.tsx

```diff
@@ -0,0 +1,67 @@
+import { Center, Flex, Spacer, Text } from '@chakra-ui/react';
+import { memo, useEffect } from 'react';
+import { useContext, useContextSelector } from 'use-context-selector';
+import { NamedExpression, NamedExpressionField } from '../../../common/types/expression';
+import { StringLiteralType } from '../../../common/types/types';
+import { GlobalContext, GlobalVolatileContext } from '../../contexts/GlobalNodeState';
+import { TypeTag } from '../TypeTag';
+import { OutputProps } from './props';
+
+export const DirectoryOutput = memo(
+    ({ label, id, outputId, schemaId, useOutputData }: OutputProps) => {
+        const type = useContextSelector(GlobalVolatileContext, (c) =>
+            c.typeState.functions.get(id)?.outputs.get(outputId)
+        );
+
+        const { setManualOutputType, schemata } = useContext(GlobalContext);
+
+        const schema = schemata.get(schemaId);
+
+        const value = useOutputData(outputId) as string;
+
+        useEffect(() => {
+            // Run this only if this is a "starting" node
+            if (!schema.inputs.some((i) => i.hasHandle)) {
```

Comment:
I don't like this definition of a starting node. It's very implicit. Could you please make a function `isStartingNode(schema: NodeSchema)` (maybe somewhere in util, idk)? That way, you wouldn't need the comment either.

Reply:
And please use the function everywhere, obviously.